File: FureyLib/C#/Templates/State Machine/State.cs
﻿
// Template State Machine State Script
// by Kyle Furey

/// <summary>
/// A template for a new state machine state.
/// </summary>
public class State : StateBase
{
    /// <summary>
    /// Constructor (sets state machine)
    /// </summary>
    /// <param name="StateMachine"></param>
    public State(StateMachine StateMachine)
    {
        stateMachine = StateMachine;

        // Change the state type enum here!
        stateMachine.currentStateType = StateType.State;
    }

    /// <summary>
    /// Called when this state is set as the state machine's current state
    /// </summary>
    public override void OnStateEnter()
    {
        // Calls the base class's function
        base.OnStateEnter();
    }

    /// <summary>
    /// Called every frame while this state is the state machine's current state
    /// </summary>
    public override void OnStateUpdate()
    {
        // Calls the base class's function
        base.OnStateUpdate();
    }

    /// <summary>
    /// Called when this state machine's current state is no longer this state
    /// </summary>
    public override void OnStateExit()
    {
        // Calls the base class's function
        base.OnStateExit();
    }
}

File: FureyLib/Python/Templates/State Machine/state.py
# Template State Machine State Script
# by Kyle Furey

from state_machine import StateBase, StateMachine


class State(StateBase):
    """A template for a new state machine state."""

    def __init__(self, state_machine: StateMachine):
        """Constructor (sets state machine)"""

        self.state_machine = state_machine

        pass

    def on_state_enter(self):
        """Called when this state is set as the state machine's current state"""

        # Calls the base class's function
        super().on_state_enter()

        pass

    def on_state_update(self):
        """Called every frame while this state is the state machine's current state"""

        # Calls the base class's function
        super().on_state_update()

        pass

    def on_state_exit(self):
        """Called when this state machine's current state is no longer this state"""

        # Calls the base class's function
        super().on_state_exit()

        pass

    pass
